### ISSUE TITLE
Review another state management

### ISSUE THREAD

**Issue by sergeysova** [react-spring/zustand#68]
Please, review https://github.com/zerobias/effector and https://github.com/artalar/reatom

**Comment by alidcastano** [comment]
came upon this repo: https://github.com/dai-shi/will-this-react-global-state-work-in-concurrent-mode

@JeremyRH looks like zustand's event emitter implementations will cause problems in concurrent mode? also, curious, is there a reason you guys decided against using context API with proxies?

**Comment by drcmda** [comment]
shallow copies have subtle but interesting properties, for instance knowing from the root up which nodes have changes till the very last atomic. i always felt like it's a very simple but powerful concept that transcends languages - and if someone wants proxies: immer.

as for tearing, i've read up on it and bumped into the following written by @markerikson

> Part of the idea behind using createContext for v6 was that since React ensures a given render pass uses the same context value everywhere, there would be no chance of tearing. The v6 implementation does work, but it's not as efficient in some cases as we'd hoped. We're currently working on coming up with a different internal implementation that goes back to using direct subscriptions instead. This does potentially mean that tearing is a possibility again, but at this point we need to sit back and wait for the React team to finish putting Concurrent Mode together before we can spend time seeing what the issues really are.

i like the levelheaded approach. i think it's fine to go and look what the problem could be, but i think it's too early to panic.

**Comment by JeremyRH** [comment]
Creating a deep copy of the state, storing it in a ref, and updating it in `useEffect` solves the tearing issue. The problem is creating a deep copy of the state can be expensive and would have to be done in every call to the state setter or in `useEffect`. The issue that has me more concerned is "check4: proper update after interrupt". I haven't figured out how to solve this without using a context provider.

**Comment by dai-shi** [comment]
Closing this as it's stale. Please file a new issue for further discussion.
